torch/_inductor/codegen/triton.py

```diff
@@ -151,7 +155,42 @@ class TritonOverrides(OpOverrides):
     """Map element-wise ops to Triton"""
 
     @staticmethod
-    def to_dtype(x, dtype: torch.dtype):
+    def to_dtype(x, dtype: torch.dtype, src_dtype: torch.dtype = None):
+        def _get_min_elements_per_thread(
+            src_dtype: torch.dtype, dst_dtype: torch.dtype
+        ) -> int:
+            # fp8 data type conversions has min_elements_per_thread requirements.
```

Comment:
Is there an explanation for why this is the case? 🤔

Reply:
Check the intrinsics in this file (also documented in the comment below): https://github.com/openai/triton/blob/10f59d8ce04052521c1bc0cb3a3f8b98918fc7e3/lib/Conversion/TritonGPUToLLVM/ElementwiseOpToLLVM.cpp#L10. It uses b32 and e4m3x2.

Reply:
Or err, I saw the link, I'm just trying to get a better understanding for why dtype conversions would have min_elements_per_thread requirements.

Reply:
I think it's because ptx cvt only provides fp8*2 intrinsics, which are used by Triton: https://docs.nvidia.com/cuda/parallel-thread-execution/index.html#data-movement-and-conversion-instructions-cvt.
However, I also don't understand why the Triton implementation for fp8_e5m2 requires 4 element per thread. Asked in the Slack channel but didn't get an answer ^^.